modules/gapi/src/streaming/onevpl/data_provider_defines.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef GAPI_STREAMING_ONEVPL_DATA_PROVIDER_DEFINES_HPP
+#define GAPI_STREAMING_ONEVPL_DATA_PROVIDER_DEFINES_HPP
+
+#ifdef HAVE_ONEVPL
+#include <vpl/mfxcommon.h>
+#include <vpl/mfxvideo.h>
+#endif // HAVE_ONEVPL
+
+#include <opencv2/gapi/own/assert.hpp>
+#include <opencv2/gapi/streaming/onevpl/data_provider_interface.hpp>
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+#ifdef HAVE_ONEVPL
+struct IDataProvider::mfx_bitstream : public mfxBitstream {};
+#else // HAVE_ONEVPL
+struct IDataProvider::mfx_bitstream {
+    mfx_bitstream() {
+        GAPI_Assert(false && "Reject to create `mfxBitstream` because library compiled without VPL/MFX support");
+    }
+};
+#endif // HAVE_ONEVPL
```

Comment:
should the whole file body be under one big `#ifdef`?

Reply:
In `NO_VPL` this whole `streaming/onevpl` directory is excluded isn't it?

Reply:
actually no - it activates some stub-like public classed which throws assert on using
here is just wrapper on VPL bitstream which figured as public interface item